ir/instr.cpp

```diff
@@ -212,15 +214,25 @@ StateValue BinOp::toSMT(State &s) const {
   case Xor:
     val = a ^ b;
     break;
+
+  case Cttz:
+    val = a.cttz();
+    not_poison &= !(a == expr::mkUInt(0, a.bits()) && b == 1u);
```

Comment:
can we write this as e.g. b == 0 || a != 0? I think it's more readable (plus a smaller formula).
Also, could you send a patch for LLVM's LangRef.html to switch it officially to poison?

Reply:
regarding the poison switch -- this should be done for ctlz at the same time, since these intrinsics should be symmetrical in this respect

Reply:
yes, that would be great!